src/lib.rs

```diff
@@ -290,17 +298,49 @@ impl Build {
             .unwrap_or_else(|| env::var("TARGET").expect("TARGET must be set"))
     }
 
-    fn find_nasm(&mut self) -> PathBuf {
-        match self.nasm.clone() {
-            Some(path) => path,
-            None => {
-                let nasm_path = PathBuf::from("nasm");
-                match is_nasm_new_enough(&nasm_path) {
-                    Ok(_) => nasm_path,
-                    Err(version) => {
-                        panic!("This version of NASM is too old: {}", version);
-                    },
+    /// Returns version string if nasm is too old,
+    /// or error message string if it's unusable.
+    fn is_nasm_new_enough(&self, nasm_path: &Path) -> Result<(), String> {
+        match get_output(Command::new(nasm_path).arg("-v")) {
+            Ok(version) => {
+                if version.contains("NASM version 0.") {
+                    Err(version)?
+                } else if let Some((major, minor, micro)) = self.min_version {
+                    let ver: Vec<u8> = version
+                        .split(" ")
+                        .skip(2)
+                        .next()
+                        .map(|ver| {
+                            ver.split(".")
+                                .map(|v| v.parse().expect("Invalid version component"))
+                                .collect()
+                        })
+                        .expect("Invalid version");
+                    if ver.len() < 3 {
+                        panic!("Not enough version components");
+                    } else {
+                        eprintln!("{:?} {:?} {:?}", self.min_version, version, ver);
+                        if major > ver[0] ||
+                        (major == ver[0] && minor > ver[1]) ||
+                         (major == ver[0] && minor == ver[1] && micro > ver[2]) {
+                            Err(version)?
+                        } else {
+                            Ok(())
+                        }
+                    }
+                } else {
+                    Ok(())
                 }
+            }
+            Err(err) => Err(err)?,
```

Comment:
All this can probably be unrolled using the `?` operator, rather than all these nested if statements. Several test cases, each of which returns `Err` if it fails.